src/f32/quat.rs

```diff
@@ -358,6 +358,64 @@ impl Quat {
         }
     }
 
+    /// Performs a spherical linear interpolation between `self` and `other`
+    /// based on the value `s`.
+    ///
+    /// When `s` is `0.0`, the result will be equal to `self`.  When `s`
+    /// is `1.0`, the result will be equal to `other`.
+    #[inline]
+    pub fn slerp(self, end: Self, s: f32) -> Self {
+        // http://number-none.com/product/Understanding%20Slerp,%20Then%20Not%20Using%20It/
+
+        glam_assert!(self.is_normalized());
+        glam_assert!(end.is_normalized());
+        
+        const DOT_THRESHOLD: f32 = 0.9995;
+
+        let dot = self.dot(end);
+
+        if dot > DOT_THRESHOLD {
+            // assumes lerp returns a normalized quaternion
+            self.lerp(end, s)
+        } else {
+            #[cfg(not(vec4sse2))]
```

Comment:
use `#[cfg(vec4f32)]`, at some point I hope to support other architectures so `#[cfg(not(vec4sse2))]` would be incorrect once that happens.